app/src/lib/stores/app-store.ts

```diff
@@ -5568,6 +5577,70 @@ export class AppStore extends TypedBaseStore<IAppState> {
     }
     return repository
   }
+
+  /**
+   * Create a tutorial repository using the given account. The account
+   * determines which host (i.e. GitHub.com or a GHES instance) that
+   * the tutorial repository should be created on.
+   *
+   * @param account The account (and thereby the GitHub host) under
+   *                which the repository is to be created created
+   */
+  public async _createTutorialRepository(account: Account) {
+    try {
+      await this.statsStore.recordTutorialStarted()
+
+      const name = 'desktop-tutorial'
+      const path = Path.resolve(getDefaultDir(), name)
+
+      const apiRepository = await createTutorialRepository(
+        account,
+        name,
+        path,
+        this.setCreateTutorialProgress
+      )
+
+      await this._addTutorialRepository(path, account.endpoint, apiRepository)
+      await this.statsStore.recordTutorialRepoCreated()
+    } catch (err) {
+      sendNonFatalException('tutorialRepoCreation', err)
+
+      if (err instanceof GitError) {
+        this.emitError(err)
+      } else {
+        this.emitError(
+          new Error(
+            `Failed creating the tutorial repository.\n\n${err.message}`
+          )
+        )
+      }
+    } finally {
+      this.setCreateTutorialRepositoryProps({ progress: undefined })
+      this._closePopup(PopupType.CreateTutorialRepository)
+    }
+  }
+
+  private setCreateTutorialProgress = (
+    title: string,
+    value: number,
+    description?: string
+  ) => {
+    this.setCreateTutorialRepositoryProps({
+      progress: { kind: 'generic', title, value, description },
+    })
+  }
+
+  private setCreateTutorialRepositoryProps(
+    state: Partial<ICreateTutorialRepositoryPopupProps>
+  ) {
```

Comment:
it looks like we're only using this to modify the `progress` property on the popup state. i believe with this typing though, you could use this method to modify the `account` property as well, which seems ill-advised. perhaps we could move `account` out of `ICreateTutorialRepositoryPopupProps` or narrow this method's typing to only modify `progress`?

(i like this as a general pattern for updating popups, just feels a little dangerous here)

Reply:
Good catch, I initially intended to port over the `loading` state to the popup props as well but I ended up deriving that state from the progress instead. I'll simplify this down to just progress.